same-origin-explainer.md

```diff
@@ -0,0 +1,119 @@
+{
+  "prerender": [
+    {"source": "list", "urls": ["https://a.test/foo"]}
+  ]
+}
+</script>
+```
+
+This hints to prerender `https://a.test/foo`.
+
+As earlier mentioned, this feature is currently restricted to same-origin URLs. See [Prerendering, revamped](README.md) for considerations on how to later support cross-origin prerendering in a privacy-preserving way.
+
+## `document.prerendering`
+
+A document can tell it is being prerendered with `document.prerendering`. Upon activation, a `prerenderingchange` event is dispatched on the document.
+
+Example:
+```javascript
+console.log(document.prerendering);  // true
+
+document.addEventListener('prerenderingchange', (event) => {
+  console.log(document.prerendering);  // false
+});
+```
+
+## Storage and cookies
+
+The prerendered page has the same access to storage and cookies as a normal page. In particular, the prerendered request includes cookies and the Set-Cookie response header modifies cookies. Storage APIs such as Indexed DB and LocalStorage also function in a prerendered page.
+
+### Session Storage
+
+Session storage is a special case. Session storage is intended to be restricted to a tab, but allowing a prerendering page to access its tab's session storage may cause breakage for sites that expect only one page capable of accessing the tab's session storage at a time. Therefore a prerendered page starts out with a clone of the tab's session storage state when it is created. Upon activation, the prerendered page's clone is discarded, and again the tab's main storage state is used instead. Pages that use session storage can use the `prerenderingchange` event to detect when this swapping of state occurs.
+
+See [this discussion](https://github.com/whatwg/storage/issues/119) for more rationale about this design.
+
+## Restricted APIs
+
+Some restrictions apply to prerendered pages, to prevent intrusive behaviors such as popping up windows, playing audio, etc.
+
+Many APIs are simply deferred in a prerendered page, that is, the processing does not occur until activation. If the API is promise-based, the promise is not resolved until activation. Some examples of these APIs include Geolocation, Web Serial, Notifications, Web MIDI, and Idle Detection.
+
+It is worth noting that APIs gated on user interaction, system focus, or visibilty are implicitly restricted in a prerendered page, since prerendered pages never have those properties. This includes `window.open()`, `element.requestFullscreen()`, and more.
+
+See the [Preventing intrusive behaviors](https://jeremyroman.github.io/alternate-loading-modes/#intrusive-behaviors) section of the specification for a list of restrictions, while this is still a work-in-progress.
+
+The user agent may also discard a prerendered page at any time, and some implementations may do this when they do not yet support deferring or handling an API in a graceful manner. For example, Chromium's implementation discards prerendered pages that use plugins.
```

Comment:
Just curious, by this I guess you mean PDFs? That's the only plugin we support.

Reply:
Hm, I'm not actually sure. Code-wise, we cancel when a "PepperHost" is requested by the renderer process. This code is only reached when the ENABLE_PLUGINS buildflag is enabled. A testcase for this uses `<embed id="testPlugin" type="application/chromium-test-default-plugin" width="42" height="42"></embed>` to cause cancellation.

I guess any "embed/object" that Chromium thinks is a plugin gets here. i'm not sure if the type is special but searching for `chromium-test-default-plugin` in codesearch doesn't bring up anything.

I'll see if I can find a better example later. I'm also thinking of adding an "Implementation-specific quirks" section for any significant Chromium behaviors that might not be intended to be specced.